pypit/arload.py

```diff
@@ -206,95 +206,125 @@ def generate_updates(dct, keylst, keys, whddict, headarr):
         if msbias is not None:
-            if type(msbias) is np.ndarray:
-                temp -= msbias  # Subtract the master bias frame
-            elif isinstance(msbias,basestring):
-                if msbias == "overscan":
-                    arproc.sub_overscan(temp, det)
-                else:
-                    msgs.error("Could not subtract bias level when loading {0:s} frames".format(frametype))
-            if trim:
-                temp = arproc.trim(temp, det)
-        # Save image
+            temp = arproc.bias_subtract(temp, msbias, numamplifiers=numamplifiers,
+                                        datasec=datasecs, oscansec=oscansecs)
+
+        if trim:
+            # Trim
+            temp = arproc.trim(temp, numamplifiers, datasecs)
+
+        # Save
         if i == 0:
-            frames = np.zeros((temp.shape[0], temp.shape[1], np.size(ind)))
+            frames = np.zeros((temp.shape[0], temp.shape[1], len(ind)))
             frames[:,:,i] = temp.copy()
         else:
             frames[:,:,i] = temp.copy()
         del temp
 
-#	pool = mpPool(processes=np.min([settings.argflag['run']['ncpus'],np.size(ind)]))
-#	async_results = []
-#	for i in range(np.size(ind)):
-#		async_results.append(pool.apply_async(fits.getdata, (fitsdict['directory'][ind[i]]+fitsdict['filename'][ind[i]], settings.spect['fits']['dataext'])))
-#	pool.close()
-#	pool.join()
-#	map(ApplyResult.wait, async_results)
-#	for j in range(np.size(ind)):
-#		if j == 0:
-#			temp = async_results[j].get()
-#			frames = np.zeros((temp.shape[0], temp.shape[1], np.size(ind)))
-#			if msbias is None:
-#				frames[:,:,i] = temp
-#			else:
-#				frames[:,:,i] = temp - msbias
-#			del temp
-#		else:
-#			if msbias is None:
-#				frames[:,:,i] = async_results[j].get()
-#			else:
-#				frames[:,:,i] = async_results[j].get() - msbias
-    if np.size(ind) == 1:
-        msgs.info("Loaded {0:d} {1:s} frame successfully".format(np.size(ind), frametype))
+    # Finish
+    if len(ind) == 1:
+        msgs.info("Loaded {0:d} {1:s} frame successfully".format(len(ind), frametype))
     else:
-        msgs.info("Loaded {0:d} {1:s} frames successfully".format(np.size(ind), frametype))
+        msgs.info("Loaded {0:d} {1:s} frames successfully".format(len(ind), frametype))
     return frames
 
 
+def load_raw_frame(spectrograph, raw_file, det, dataext01=None, disp_dir=0):
+    """
+    Load data frames, usually raw.
+    Bias subtract (if not msbias!=None) and trim (if True)
+
+    Parameters
+    ----------
+    raw_file : str
+       Full path to raw_file
+    det : int
+      Detector number requested, starts at 1
+    disp_dir : int, optional
+      if 1, Transpose the image to align spectral dimension with columns
+
+    Returns
+    -------
+    frame : ndarray
+      the raw_frame
+    """
+    def load_indfr(name,ext):
+        msgs.work("Trim and overscan has not been applied")
+        temp = fits.getdata(name, ext)
+        return temp
+
```

Comment:
I'm not a huge fan of defining functions within other functions.  In this case, is `load_indfr` actually used?